src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -411,6 +413,60 @@ def _generate_thread_status_messages(self, pool: ThreadPool, total_count: int):
             msgs.append("Processing Lines: " + ", ".join(lines) + ".")
         return msgs
 
+    def _determine_worker_count(self):
```

Comment:
We can then update the logic of this method into something like:
```python
def _determine_worker_count(self):
    worker_count = self._load_worker_count_from_env():
    if worker_count:
        # If worker count is specified, use the value as-is.
        return worker_count

    worker_count = min(self._DEFAULT_WORKER_COUNT, self._nlines)
    
    if not self._use_fork:
        # Starting a new process in non-fork mode requires to allocate memory. 
        # Determine the maximum number of processes based on available memory
        # to avoid memory bursting.
        estimated_available_worker_count = get_available_max_worker_count()
        worker_count = min(worker_count, estimated_available_worker_count)

    return worker_count
```

Reply:
fixed

Reply:
fixed